src/app/data/typeChart.ts

```diff
@@ -132,23 +146,30 @@ export function calcTypeMatchup(atk: AttackerData, def: DefenderData) {
             atkAbilityCalc = 1.3;
         }
     }
-    const atkMove = atk.move;
-    if (atkMove !== undefined) {
-        const doubleDealtMatch = doubleDealtMoves.find((x) => x.move == atkMove.id);
+
+    let atkMoveCalc = 1;
+    if (atk.move != undefined && !isNull(atk.move) && atk.move.isAttackingMove()) {
+        const doubleDealtMatch = doubleDealtMoves.find((x) => x.move == atk.move!.id);
```

Comment:
It's probably not surprising that at some point in the future I want to replace these lists of special moves with move subclasses like ExtraTypeMove, by checking function codes, like I am with all other move functionality. But that doesn't have to be part of this PR, I'll get to them when they come up on the move spreadsheet.